### PR TITLE
Dockerfile: Cleanup, copy libgfortran deps

### PR DESCRIPTION
**Pull request by dpipemazo**
- libgfortran was being copied to /usr/local/lib, but we were
not copying over its dependencies as well. Now, when we copy
libgfortran, we also will copy anything it depends on to ensure
it's all present in the output build
- The error above would manifest as numpy not being able to load

## Change description

> Description here

## Type of change
- [ ] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review 

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -48,9 +65,12 @@ RUN apt-get update \
...
    pkg-config
 
-#we need to copy libgfortran from /usr/lib/gcc/arch-linux-gnu to
-#/usr/local/lib so python tests could find them
+# We need to copy libgfortran and all of its dependencies from /usr/lib/gcc/arch-linux-gnu to
+# /usr/local/lib so that they get packaged up with the system install
 RUN cp `dirname \`find /usr/lib -iname libgfortran.so.4 -not -path /usr/lib/gcc\``/libgfortran* /usr/local/lib 
+RUN ldd /usr/local/lib/libgfortran* | grep "=> /" | awk '{print $3}' | sort -u > /tmp/required_libs.txt
+COPY --from=no-deps /tmp/existing_libs.txt /tmp/existing_libs.txt
+RUN diff --new-line-format="" --unchanged-line-format="" /tmp/required_libs.txt /tmp/existing_libs.txt | grep -v /usr/local/lib > /tmp/libs_to_copy.txt && xargs -a /tmp/libs_to_copy.txt cp -L -t /usr/local/lib || exit 0
```
Comment (aywaldron) [inline Dockerfile:73]: While we're in here we could add a comment explaining what this line does?

File: Dockerfile [hunk 2]
```diff
@@ -13,6 +13,23 @@ ARG STOCK_IMAGE=ubuntu:bionic-20210827
 ARG ATOM_BASE=atom-base
 ARG DEBIAN_FRONTEND=noninteractive
 
+################################################################################
+#
+# no-deps: Stock image with list of pre-existing libraries
+#
+################################################################################
+FROM $STOCK_IMAGE as no-deps
```
Comment (aywaldron) [inline Dockerfile:21]: :+1: definitely more clear to have this up here

File: Dockerfile [hunk 3]
```diff
@@ -324,7 +331,7 @@ ENV DISPLAY :0
 
 ################################################################################
 #
-# Base image: atom + CV tools + CUDA tools
+# atom-base-cuda: atom + CV tools + CUDA tools
```
Comment (aywaldron) [inline Dockerfile:334]: Thanks for updating these names

**Review by aywaldron — APPROVED** [review]
Thanks for getting this fix up Dan!

**Comment by dpipemazo** [comment]
@aywaldron I've packaged up the library copy functionality into some helper scripts in `utilities/docker` and also cleaned up the issue where the CircleCI job names were not showing up properly in the build checks due to the `/` in the name. Hopefully once this all passes we should be good to go.

**Comment by aywaldron** [comment]
@dpipemazo Awesome!

**Pull request merged**